coverage.c

```diff
@@ -609,20 +612,28 @@ int main_coverage(int argc, char *argv[]) {
             for (j = 0; j < n_plp[i]; ++j) {
                 const bam_pileup1_t *p = plp[i] + j; // DON'T modify plp[][] unless you really know
 
-                if (p->is_del || p->is_refskip) --depth_at_pos; // having dels or refskips at tid:pos
-                else if (p->qpos < p->b->core.l_qseq &&
-                        bam_get_qual(p->b)[p->qpos] < opt_min_baseQ) --depth_at_pos; // low base quality
-                else
-                    stats[tid].summed_baseQ += bam_get_qual(p->b)[p->qpos];
+                if (p->is_del || p->is_refskip) {
+                    --depth_at_pos; // having dels or refskips at tid:pos
+                } else if (p->qpos < p->b->core.l_qseq) {
+                    if (bam_get_qual(p->b)[p->qpos] < opt_min_baseQ) {
+                        --depth_at_pos; // low base quality
+                    } else {
+                        stats[tid].summed_baseQ += bam_get_qual(p->b)[p->qpos];
```

Comment:
While not new logic, this means that the "Mean BaseQ" reported by the tool is the average quality of bases which meet the minimum base quality criteria, rather than the average quality of all bases.  This isn't what I imagined it to be and the man page is rather ambiguous given it doesn't explicitly lean one way or the other.

It's probably too late to change the program functionality, unless it's a clear bug (but `depth_at_pos` is also filtered depth).  The man page problem just needs some tweakage to explain that the filtering takes place before computing averaged statistics.

Reply:
Alignments that don't have the minimum base quality don't get included in the coverage calculations.  So it is the average quality of used alignments.  I think the man page covers that.

This change does mess with the calculations, hence the warning.  It is a compromise so that the program doesn't have to exit when it comes across a missing quality value.